tasks/release.py

```diff
@@ -533,17 +559,131 @@ def _get_highest_version_from_release_json(release_json, highest_major, version_
+    - the tag must match compatible_version_re (the main usage is to restrict the compatible tags to the
+      ones with the same minor version as the Agent)?
+    
+    If check_for_rc is true, a warning will be emitted if the latest version that satisfies
+    the constraints is an RC. User confirmation is then needed to check that this is desired.
+    """
+
+    version = _get_highest_repo_version(github_token, repo_name, "", allowed_major_versions, compatible_version_re)
+    if not version:
+        raise Exit(TAG_NOT_FOUND_TEMPLATE.format(repo_name, _stringify_version(new_agent_version)), 1)
+    if check_for_rc and version["rc"] is not None:
+        if not yes_no_question(
+            RC_TAG_QUESTION_TEMPLATE.format(repo_name, _stringify_version(version)), "orange", False,
+        ):
+            raise Exit("Aborting release.json update.", 1)
+
+    print(TAG_FOUND_TEMPLATE.format(repo_name, _stringify_version(version)))
+    return version
+
+
+def _confirm_independent_dependency_repo_version(repo, latest_version, highest_release_json_version):
+    """
+    Checks if the two versions of a repository we found (from release.json and from the available repo tags)
+    are different. If they are, asks the user for confirmation before updating the version.
+    """
+
+    confirmed_version = highest_release_json_version
+    if latest_version != highest_release_json_version:
+        print(
+            color_message(
+                DIFFERENT_TAGS_TEMPLATE.format(
+                    repo, _stringify_version(latest_version), _stringify_version(highest_release_json_version)
+                ),
+                "orange",
+            )
+        )
+        if yes_no_question(
+            "Do you want to update {} to {}?".format(repo, _stringify_version(latest_version)), "orange", False
+        ):
+            confirmed_version = latest_version
+
+    return confirmed_version
+
+
+def _fetch_independent_dependency_repo_version(
+    repo_name, release_json, agent_major_version, github_token, release_json_key
 ):
+    """
+    Fetches the latest tag on a given repository whose version scheme doesn't match the one used for the Agent:
+    - first, we get the latest version used in release entries of the matching Agent major version
+    - then, we fetch the latest version available in the repository
+    - if the above two versions are different, emit a warning and ask for user confirmation before updating the version.
+    """
+
+    highest_version = _get_highest_version_from_release_json(
+        release_json, agent_major_version, VERSION_RE, release_json_key=release_json_key,
+    )
+    # NOTE: This assumes that the repository doesn't change the way it prefixes versions.
+    version = _get_highest_repo_version(github_token, repo_name, highest_version["prefix"], None, VERSION_RE)
+    if not version:
+        raise Exit("Couldn't find any {} version.".format(repo_name), 1)
```

Comment:
Correct me if I am wrong but I think every time we call `_get_highest_repo_version` we check if the output is None and then raise an exception. Should we do that inside the function?